### ISSUE TITLE
UD MLX5 failure

### ISSUE THREAD

**Issue by shamisp** [openucx/ucx#1470]
```
ud_ep.c:496  Assertion `ctl->type == UCT_UD_PACKET_CREP' failed
```

@yosefe please talk with me for details

**Comment by shamisp** [comment]
```
[----------] 1 test from ud_mlx5/test_uct_perf
[ RUN      ] ud_mlx5/test_uct_perf.envelope/0
  ud_ep.c:496  Assertion `ctl->type == UCT_UD_PACKET_CREP' failed
```

**Comment by shamisp** [comment]
on the same setup in debug mode:
```
1 test from udx/test_ucp_perf
[ RUN      ] udx/test_ucp_perf.envelope/0
ud_mlx5.c:395  Assertion `ntohs(cqe->wqe_counter) == iface->rx.wq.cq_wqe_counter' failed
```

**Comment by shamisp** [comment]
fixed by #1527 #1558
